### PR TITLE
LOG-5154: Add durable queue to outputs for AtLeastOnce delivery

### PR DESCRIPTION
**Pull request by jcantrill**
### Description
This PR:
* Adds disk buffering to act as the persistent queue until kubernetes_logs supports acknowledgements
* Configures the minimum required disk
* Introduces an e2e test for AtLeastOnce
* blocked by #2383 
* Fixes duplicate log collection on restart

### Links
* https://issues.redhat.com/browse/LOG-5154
* https://issues.redhat.com/browse/LOG-5156

cc @cahartma @Clee2691

### CODE REVIEW

File: internal/generator/vector/output/config_strategy.go [hunk 1]
```diff
@@ -53,6 +56,8 @@ func (o Output) VisitBuffer(b common.Buffer) common.Buffer {
 		switch o.spec.Tuning.Delivery {
 		case logging.OutputDeliveryModeAtLeastOnce:
 			b.WhenFull.Value = common.BufferWhenFullBlock
+			b.Type.Value = buffertTypeDisk
+			b.MaxSize.Value = minBufferSize
```
Comment (vparfonov) [inline internal/generator/vector/output/config_strategy.go:60]: is it correct, **Max**Size set to **min**BufferSize?

Comment (jcantrill) [reply internal/generator/vector/output/config_strategy.go:60]: yes.  we are not allowing users to actually set the on disk buffer size and 'minBufferSize' is the value required to use on disk buffering

File: test/e2e/collection/tuning/streams.go [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+}
+
+func (ls LogStreams) Evaluate() {
+	log.V(1).Info("Evaluating log streams", "total", len(ls))
+	for name, s := range ls {
+		if len(s.SequenceIDs) > 0 {
+			sort.Ints(s.SequenceIDs)
+			log.V(3).Info("Messages received", "name", name, "tot", len(s.SequenceIDs), "minID", s.SequenceIDs[0], "maxID", s.SequenceIDs[len(s.SequenceIDs)-1])
+			s.First = s.SequenceIDs[0]
+			s.Last = s.SequenceIDs[len(s.SequenceIDs)-1]
```
Comment (vparfonov) [inline test/e2e/collection/tuning/streams.go:56]: nit: we can calculate it once

**Pull request merged**